frontend/src/pages/ArtifactDetails.tsx

```diff
@@ -61,6 +62,16 @@ export default class ArtifactDetails extends Page<{}, ArtifactDetailsState> {
     this.load = this.load.bind(this);
   }
 
+  componentDidUpdate(prevProps: Readonly<{} & PageProps>, prevState: Readonly<ArtifactDetailsState>, snapshot?: any): void {
+    if (this.props.match.params[RouteParams.ID] !== prevProps.match.params[RouteParams.ID]) {
```

Comment:
_**Nit**: Invert condition to early return (decrease code depth)_
```suggestion
    if (this.props.match.params[RouteParams.ID] === prevProps.match.params[RouteParams.ID]) return
```

Reply:
Done